src/code/actions.js

```diff
@@ -295,25 +295,23 @@ function _rejectEggFromBasket(eggIndex, basketIndex) {
   };
 }
 
-export function rejectEggFromBasket() {
-  return (dispatch, getState) => {
-    const { submittedEggIndex, submittedBasketIndex } = getState();
-    dispatch(_rejectEggFromBasket(submittedEggIndex, submittedBasketIndex));
+export function rejectEggFromBasket(args) {
+  return (dispatch) => {
+    dispatch(_rejectEggFromBasket(args.eggDrakeIndex, args.basketIndex));
   };
```

Comment:
You're right that I could have eliminated the thunk when I switched from using state to using args. In this case, I decided to leave it because I have a suspicion that when I implement the upcoming coin story that those two action creators will be dispatching the `CHALLENGE_COMPLETE` action as well.